app/knight.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 weapon: dict,
+                 armour: list,
+                 potion: dict) -> None:
+        self.name = name
+        self.power = power + weapon["power"]
+        self.hp = hp
+        self.weapon = weapon
+        self.protection = sum(element["protection"] for element in armour)
+        if potion is not None:
+            self.apply_potion_effects(potion)
+
+    def apply_potion_effects(self, potion: dict) -> None:
+        effects = potion.get("effect", {})
+        self.power += effects.get("power", 0)
+        self.hp += effects.get("hp", 0)
+        self.protection += effects.get("protection", 0)
+
+    @staticmethod
+    def fight(knight1: Knight, knight2: Knight) -> None:
+        if knight1.protection < knight2.power:

```

Comment:
No need this if. Remove it
Also with line 31

Reply:
Done